### PR TITLE
fixes #1783

### LINKED ISSUE #1783: Use KEYWORD_ONLY parameter in electron func will cause `ValueError`
### Environment

- **Covalent version**: 0.220.0
- **Python version**: 3.10.11
- **Operating system**: Ubuntu 22.04.1 LTS

### What is happening?

When KEYWORD_ONLY parameter in `func`, calling `ct.electron(func)` will give a `ValueError`.

### How can we reproduce the issue?

```python
@ct.lattice
@ct.electron
def test_func(a, *, b):
    return a + b

ct.dispatch(test_func)(a=1, b=2)
```

### What should happen?

This example will raise `ValueError: Unexpected keyword arguments: b`

### Any suggestions?

https://github.com/AgnostiqHQ/covalent/blob/acfc902e2296bdf6b9210789516e12b9fd1b94a2/covalent/_shared_files/utils.py#L218-L221

To change these codes as follows, maybe solve this problem.

```python
elif param.kind == param.VAR_KEYWORD:
    for key, value in kwargs.items():
        if key != param_name:
            named_kwargs[key] = value
elif param.kind == param.KEYWORD_ONLY:
    if param_name in kwargs:
        named_kwargs[param_name] = kwargs[param_name]
```

### PR DESCRIPTION
**Pull request by AdityaRaj23**
- [ ] I have added the tests to cover my changes.
- [ ] I have updated the documentation and CHANGELOG accordingly.
- [x] I have read the CONTRIBUTING document.

### CODE REVIEW

**Comment by kessler-frost** [comment]
Hey @AdityaRaj23! Thanks for this, you might wanna update the CHANGELOG.md file briefly explaining what did you change under the appropriate section, which for this one is the "Fixed" section, also can you remove the yarn.lock change that was pushed in this one as well. Let me know if you need any help anywhere.

**Comment by AdityaRaj23** [comment]
should i leave the code i changed commented or remove it before making another pull request?

**Comment by kessler-frost** [comment]
You don't have to make another pull request, you can just commit and push to this branch itself and it'll be updated here.

**Comment by kessler-frost** [comment]
@AdityaRaj23 , it seems like you have deleted the `yarn.lock` file which isn't what I quite meant, I meant that it shouldn't be different than what it currently is in our `develop` branch.

One tip for the commit messages is that they should be a bit more informative as to what was done in that particular commit. Making only one commit with `fixed AgnostiqHQ#1783` is sufficient to connect this PR to the 1783 issue.

**Comment by AdityaRaj23** [comment]
Sorry, I read the contributing.md there it was written ,for fixing you should commit with message 'fixes#{issue no}' so i stick to that but will surely keep this in my mind from now onwards for any commit

File: CHANGELOG.md [hunk 1]
```diff
@@ -5042,3 +5042,9 @@ Installed executor plugins don't have to be referred to by their full module nam
 - CHANGELOG.md to track changes (this file).
 - Semantic versioning in VERSION.
 - CI pipeline job to enforce versioning.
+
+
+### Fixed
+
+- Fixed the bug that caused ValueError error when using KEYWORD_ONLY parameter in electron func 
+- changed code at line 218 in covalent/_shared_files/utils.py
```
Comment (kessler-frost) [inline CHANGELOG.md:5050]: This needs to be near the top of the file under the `[UNRELEASED]` heading. You might see there's already a `Fixed` section within which you can add these.

File: covalent_ui/webapp/yarn.lock
Comment (kessler-frost) [inline covalent_ui/webapp/yarn.lock:1]: You seem to have deleted this file, copying this file from `develop` to your branch to its respective place and pushing again will make sure this hasn't been modified unnecessarily.

**Review by kessler-frost — CHANGES_REQUESTED** [review]

**Comment by kessler-frost** [comment]
Hey @AdityaRaj23 , seems like the test coverage isn't up to the mark with what is expected, so you'll need to add a test for the changes that you made. This will also make sure that the thing that you fixed will not be broken by any of the later commits by anyone else.

So, the file where you'll wanna add the change is `tests/covalent_tests/shared_files/utils_test.py`, and the function that needs to be tested is `get_named_params`. A simpler way to reproduce the same error above is:
```python
from covalent._shared_files.utils import get_named_params

def test_func(a, *, b):
    return a + b

named_args, named_kwargs = get_named_params(test_func, [1], {'b': 2})
```
So, you essentially just need to test that this isn't failing and then the PR should be gtg.

**Comment by AdityaRaj23** [comment]
Ok will do as directed

**Comment by kessler-frost** [comment]
Hey @AdityaRaj23 just bumping this one. You're almost there and I think it would be a shame if this didn't get merged for Hacktoberfest as this is pretty much done. You just need to put the test that you wrote in a function and that would be it.

**Comment by AdityaRaj23** [comment]
> Hey @AdityaRaj23 just bumping this one. You're almost there and I think it would be a shame if this didn't get merged for Hacktoberfest as this is pretty much done. You just need to put the test that you wrote in a function and that would be it.

I would do that today positively was busy in some other work ...

**Comment by AdityaRaj23** [comment]
tests are fine on my side what is the issue with this debian 10 and debian 11 can you brief me about this i am not sure whats wrong...

**Comment by kessler-frost** [comment]
You can see the reason why the test failed here: https://github.com/AgnostiqHQ/covalent/actions/runs/6705590122/job/18231396245?pr=1810. The failing statement is `assert named_args == [1]` whereas it should be `assert named_args == {'a': 1}` instead. This is since everything will get an appropriate name from running `get_named_params`.

**Comment by kessler-frost** [comment]
@AdityaRaj23 Thanks for this, we'll get this merged now.

**Comment by kessler-frost** [comment]
Thanks @AdityaRaj23 ! I'm glad you'd like to contribute more we're always looking for eager contributors like you. Please feel free to take a look at issues tagged with `good-first-issue` or `help-wanted` labels if you're interested.

**Pull request merged**